pkgs/servers/mail/dovecot/plugins/pigeonhole/compile.nix

```
@@ -0,0 +1,38 @@
+{dovecot_pigeonhole, lib, writeText }:
+{ name, plugins ? [], extensions ? [], globalExtensions ? [], meta ? {}, ... } @ args:
+with lib;
+let
+  config = writeText "buildSieveScripts-${name}-config" ''
+    plugin {
+      ${optionalString (plugins != [])
+        "sieve_plugins = ${concatStringsSep " " (map (e: "sieve_${e}") plugins)}"}
+      ${optionalString (extensions != [])
+        "sieve_extensions = ${concatStringsSep " " extensions}"}
+      ${optionalString (globalExtensions != [])
+        "sieve_global_extensions = ${concatStringsSep " " globalExtensions}"}
+    }
+  '';
+in
+
+dovecot_pigeonhole.stdenv.mkDerivation (args // {
+  inherit name;
+
+  installPhase = ''
+    mkdir -p "$out/sieve/"
+    for k in $(find * -name \*.sieve); do
+      mkdir -p "$out/sieve/$(dirname "$k")"
+      cp -a "$k" "$out/sieve/$k"
+      if [[ ! -L "$k" ]]; then
+        echo "Compiling $k"
+        touch -m --date=@0 "$out/sieve/$k"
+      else
+        echo "Compiling $k -> $(readlink -f "$k")"
+      fi
+      ${dovecot_pigeonhole}/bin/sievec -c "${config}" "$out/sieve/$k"
```

```suggestion
     sievec -c "${config}" "$out/sieve/$k"
```

This should use nativeBuildInputs where we add  dovecot_pigeonhole.

---

Done!